[docs] Add jsonb upsert (#9852)
yugabyte/yugabyte-db: docs/content/latest/integrations/apache-spark/java.md

        .option("spark.cassandra.json.quoteValueString", "true").mode(SaveMode.Append).save();
```

Here is sample content of the Dataset:

```output
+---+-----+----------------+--------------------------------------------------------------+
|id |name |address         |phone                                                         |
+---+-----+----------------+--------------------------------------------------------------+
|5  |SAM  |INDIA           |{"call":"75675655","code":"91"}                               |
|1  |ram  |null            |{"Module":"CM","call":"932233342","code":"+42","phone":"200"}|
+---+-----+----------------+--------------------------------------------------------------+
```

#### Configuring JSONB upsert

##### spark.cassandra.json.quoteValueString
 
Specifies whether the JSONB field values should be quoted as string. Defaults to false (since this is not space efficient).
 
The option allows you to preserve floating point precision for JSONB fields. For un-quoted non-numeric values, double quotes are added. Otherwise the values would be rejected by YCQL.

Here is an example - note the quotes around 100.1:

```json

> "dl":1122,"rsrp":["abc","rsrq":null,"sinr":"100.1"]

Author: @nchandrappa

> "dl":1122,"rsrp":["abc","rsrq":null,"sinr":"100.1"]

Reviewer: Yes, let's update the JSON snipped to a well-formed one Alex has provided above. Basically, snippet is trying to show quotes around json values.

Suppose `{"dl": 5, "ul":"foo"}` is stored in the JSONB column `usage`, and `{"x": "foo", "y":"bar"}` is stored in the JSONB column `z`. The mapping would be expressed as follows:

```json
dl,ul:usage;x,y:z 
```

where `dl` and `ul` are fields in `usage`, and `x` and `y` are fields in `z`.
 
This mapping is for dataframe `save()` operations where you specify the JSONB column(s) that the given JSON fields map to. 
 
When binding individual rows, missing JSONB field values are set to `null`. This allows data in the csv file to load uninterrupted. To turn off this behaviour, use `spark.cassandra.output.ignoreNulls`.

The following example code uses mapping:

```json
Dataset<Row> updatedRows = spark
    .sql("select date as gdate,imsi,rsrp,sinr from temp ");
updatedRows.write().format("org.apache.spark.sql.cassandra").option("keyspace", KEYSPACE)
    .option("table", CDR_TABLE)
    .option("spark.cassandra.mergeable.json.column.mapping", "dl,rsrp,sinr,ul:usage")
```

##### spark.cassandra.output.ignoreNulls

Set to true to cause all null values to be left as unset rather than bound.

> For example, for `{"dl": null, "ul":"foo"}`, if `spark.cassandra.output.ignoreNulls` is specified, the value null would be omitted for field `dl` when writing (merging) into the corresponding JSONB column, as though field `dl` didn't appear in the input.

Reviewer 2: ```suggestion
For example, for `{"dl": null, "ul":"foo"}`, if `spark.cassandra.output.ignoreNulls` is specified, only the `ul` field gets written (merged) into the corresponding JSONB column, and the `ul` field is ignored.
```
